setup-local/src/binaryControl.js

```diff
@@ -0,0 +1,232 @@
+const tc = require('@actions/tool-cache');
+const io = require('@actions/io');
+const exec = require('@actions/exec');
+const core = require('@actions/core');
+const github = require('@actions/github');
+const os = require('os');
+const path = require('path');
+const fs = require('fs');
+const Utils = require('./utils');
+const ArtifactsManager = require('./artifacts');
+const constants = require('../config/constants');
+
+const {
+  BINARY_LINKS,
+  LOCAL_BINARY_FOLDER,
+  PLATFORMS,
+  LOCAL_BINARY_NAME,
+  LOCAL_LOG_FILE_PREFIX,
+  LOCAL_BINARY_TRIGGER,
+  ALLOWED_INPUT_VALUES: {
+    LOCAL_TESTING,
+  },
+  ENV_VARS: {
+    BROWSERSTACK_LOCAL_LOGS_FILE,
+  },
+} = constants;
+
+/**
+ * BinaryControl handles the operations to be performed on the Local Binary.
+ * It takes care of logs generation and triggering the upload as well.
+ */
+class BinaryControl {
+  constructor(stateForBinary) {
+    this.platform = os.platform();
+    this.stateForBinary = stateForBinary;
+
+    this._decidePlatformAndBinary();
+  }
+
+  /**
+   * decides the binary link and the folder to store the binary based on the
+   * platform and the architecture
+   */
+  _decidePlatformAndBinary() {
+    if (this.platform === PLATFORMS.DARWIN) {
+      this.binaryLink = BINARY_LINKS.DARWIN;
+      this.binaryFolder = path.resolve(process.env.HOME, 'work', 'binary', LOCAL_BINARY_FOLDER, this.platform);
+    } else if (this.platform === PLATFORMS.LINUX) {
+      this.binaryLink = os.arch() === 'x32' ? BINARY_LINKS.LINUX_32 : BINARY_LINKS.LINUX_64;
+      this.binaryFolder = path.resolve(process.env.HOME, 'work', 'binary', LOCAL_BINARY_FOLDER, this.platform);
+    } else if (this.platform === PLATFORMS.WIN32) {
+      this.binaryLink = BINARY_LINKS.WINDOWS;
+      this.binaryFolder = path.resolve(process.env.GITHUB_WORKSPACE, '..', '..', 'work', 'binary', LOCAL_BINARY_FOLDER, this.platform);
+    }
+  }
```

Comment:
should we handle else? is it possible to have other options like openbsd in case of self-hosted runners, throwing error or something in case of other options?

Reply:
default handling sort of ^. If some new platform support added by github actions.

Reply:
will do that..

Reply:
done

Reply:
resolving this.